subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -138,6 +140,24 @@ public List<String> getRuntimeOnlyDependencies() {
         }
     }
 
+    private static class CachingTestFrameworkTransformer implements Transformer<TestFramework, VersionedTestingFramework> {
+        private final Transformer<TestFramework, TestingFramework> mapToTestFramework;
+        private Pair<TestingFramework, TestFramework> currentMapping;
```

Comment:
💭 IDK if the use of `Pair` adds much here vs. just storing the input and output directly.  I was also a little confused at first when reading this since our `Transformer` puts the generic OUT param first and the IN param second, which seems backwards to me.  Might be worth a note to clarify that for anyone else expecting the opposite.

Reply:
Ah, yes, I can see why the differing order of the types might be confusing.  I've updated this to change the `Pair` to reflect the same order as `Transformer` and added a note, which hopefully makes this less confusing.  I'd like to keep it a `Pair` just to bake in the idea that these two values are related and cannot be set independently (i.e. these are a _mapping_ with an inherent relationship).  If you feel like the vagueness of a `Pair` is confusing, I don't feel super strongly about it, but I had actually started to remove it and then decided I liked it better as it is.